src/browser/atom-window.coffee

```diff
@@ -121,6 +119,9 @@ class AtomWindow
         false
 
   handleEvents: ->
+    @browserWindow.on 'close', ->
+      global.atomApplication.saveState(false)
```

Comment:
I'm not sure if this breaks anything (as it gets called when every window is closed now, not just when the entire app is closing), but it did solve the uncaught exception.